BASE/src/ServerTelemetryChannel/Implementation/Transmitter.cs

```diff
@@ -169,10 +173,84 @@ internal virtual void Enqueue(Transmission transmission)
 
             if (!this.Storage.Enqueue(transmissionGetter))
             {
+                transmission.HasFlushTask = false;
```

Comment:
Trying to understand why does this has to be set to false here... Transmitter receives a transmission..How can it determine if HasFlushTask should be true or not, as HasFlushTask depends on if it was called from Flush..

Reply:
If a move to storage fails, we set this flag. `HasFlushTask` will be honored only in case of FlushAsync call.
`Transmitter.FlushAsync` calls `Enqueue` to follow the normal process. During that attempt if a transmission object fails to move to storage that needs to be tracked hence `HasFlushTask` is used as a marker.